kickstarter/android-oss
# MBL-936 && MBL-935: Upgrade dependencies

# 📲 What

- Upgraded Segment SDK alongside Braze SDK
- Upgraded Facebook 
- Upgraded Firebase-Bom 
- Removed failed experiment with DataStore and removed dependency.

# 🤔 Why

- Getting ready for Android 14 with the latests dependencies SDK's. 
- After latest Android regression push notifications were not being displayed on Android 14 device, reason for this had nothing to do with the actual app, but from config issues on Braze staging.
- Gems updated to be able to execute locally `bundle exec fastlane internal` 

# 🛠 How
- Take a look at Braze `Messages Activity Log`
![Screenshot 2023-10-11 at 2 11 52 PM](https://github.com/kickstarter/android-oss/assets/4083656/9f7f8d96-e0ed-4df9-abd0-4132ea0e929d)
ℹ️  `sender_id_mismatch` happens when using the wrong build to test.
ℹ️  `device_unregistered` happens when using the correct internal release build, but the staging API instead of production.

# 👀 See
- Facebook login working


https://github.com/kickstarter/android-oss/assets/4083656/0a7f4dd3-6a1d-4d21-acea-6ccd104fa7b1

- Braze push notifications working on Android 14 device (Video available on slack)

| --- | --- |
|  |  |

# 📋 QA
- Log in with regular account/ Login with facebook
- Check the analytic events are piped to Segment
- Braze push will be able to be tested downloading internal release | 3.16.0 (2014150889)

# Story 📖

[MBL-936](https://kickstarter.atlassian.net/browse/MBL-936)
[MBL-935](https://kickstarter.atlassian.net/browse/MBL-935)


[MBL-936]: https://kickstarter.atlassian.net/browse/MBL-936?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
